Makefile

```diff
@@ -0,0 +1,17 @@
+.PHONY: all auto_generated clean install run
+
+all: auto_generated
+	dotnet build
+
+auto_generated:
+	./scripts/update_common_constants.rb `pwd`
+
+clean:
+	dotnet clean
+	rm -f Perlang.Common/CommonConstants.Generated.cs
+
+install: all
+	./scripts/local_install_linux.sh
+
+run:
+	cd Perlang.ConsoleApp && dotnet run
```

Comment:
`make` is clearly not the best tool in the world for handling building of software systems, but its ubiquity and bookshelves full of documentation make it useful. I've used Rake (in a fairly sophisticated manner) in other projects and I'm not so fond of it since it tends to become rather complex over time. That's not to say that Makefiles are always _easy_ but... let's at least try to keep them this way in this case.

As can be seen, I also added a few other useful targets while I was at it:

- `clean`
- `install`
- `run`